File: frontend/src/types/backup.ts
export interface Backup {
  id: string
  name: string
  base_backup_name: string
  query_id: string
  status: string // BACKUP_CREATED, BACKUP_FAILED, BACKUP_IN_PROGRESS, BACKUP_COMPLETED
  error: string
  start_time: string
  end_time: string
  num_files: number // uint64 from backend
  total_size: number // uint64 from backend, bytes
  num_entries: number // uint64 from backend
  uncompressed_size: number // uint64 from backend, bytes
  compressed_size: number // uint64 from backend, bytes
  files_read: number // uint64 from backend
  bytes_read: number // uint64 from backend, bytes
  sql_query?: string // SQL query used for backup
}

export interface BackupStatsResponse {
  total: number // uint64 from backend
  in_progress: number // uint64 from backend
  completed: number // uint64 from backend
  failed: number // uint64 from backend
}

export interface BackupListResponse {
  items: Backup[]
  pagination: BackupPagination
}

export interface BackupPagination {
  limit: number
  offset: number
  total: number // uint64 from backend
}

File: agent/internal/clickhouse/models/backup.go
package models

// Backup represents a single backup record from ClickHouse system.backups table
type Backup struct {
	ID               string `json:"id"`
	Name             string `json:"name"`
	BaseBackupName   string `json:"base_backup_name"`
	QueryID          string `json:"query_id"`
	Status           string `json:"status"` // BACKUP_CREATED, BACKUP_FAILED, BACKUP_IN_PROGRESS, BACKUP_COMPLETED
	Error            string `json:"error"`
	StartTime        string `json:"start_time"`
	EndTime          string `json:"end_time"`
	NumFiles         uint64 `json:"num_files"`
	TotalSize        uint64 `json:"total_size"` // bytes
	NumEntries       uint64 `json:"num_entries"`
	UncompressedSize uint64 `json:"uncompressed_size"` // bytes
	CompressedSize   uint64 `json:"compressed_size"`   // bytes
	FilesRead        uint64 `json:"files_read"`
	BytesRead        uint64 `json:"bytes_read"` // bytes
	SQLQuery         string `json:"sql_query"`
}
